test/modules/landing/landing_router_test.dart

```diff
@@ -0,0 +1,65 @@
+import 'package:flutter/widgets.dart';
+import 'package:flutter_test/flutter_test.dart';
+import 'package:mockito/mockito.dart';
+import 'package:quick_test/quick_test.dart';
+import 'package:survey/modules/home/home_module.dart';
+import 'package:survey/modules/landing/landing_module.dart';
+import 'package:survey/modules/login/login_module.dart';
+
+import '../../mocks/build_context.dart';
+import '../../mocks/navigator_state.dart';
+
+class FakeAnimation extends Mock implements Animation<double> {}
```

Comment:
it's a bit odd to me that we have to add this mock instance to test the animation 👀  normally, I'd recommend adjusting the test type to run as an integration test or widget test, we might validate the real behavior instead of these abstract mocking...it's just a bit a lot of legwork for the setup. On the case of emitting the value i.e `animationDidFinish` I think if we could access the value emission from the Presenter, you can manipulate the stream as you wish 🤔  just an idea, maybe you want to recheck your current architecture abstraction later.

Reply:
actually, my bad, it's extends from Mock and being generated on the mocking behavior correctly, so no concern. Maybe the definition of the `class FakeAnimation` at the start of a test class is bit odd 😓

Reply:
Agree, i think it should be `MockAnimation` lol. Let me update it 👍